### ISSUE TITLE
Using Reveal Modal in Interchange

### ISSUE THREAD

**Issue by sky1tech** [zurb/foundation#6229]
I'm on wordpress 4.1 and I can successfully get php files to interchange but can't seem to get the modal to work from within an Interchange file. 

<div data-interchange="[<?php echo get_stylesheet_directory_uri();?>/templates/socket-mobile.php, (small)], [<?php echo get_stylesheet_directory_uri();?>/templates/socket.php, (default)]"></div>

Both files begin with the following (which makes php files work correctly in wordpress):
<?php
require($_SERVER['DOCUMENT_ROOT'].'/wp-load.php');
?>

Basically we are okay with the modal loading on larger devices but due to the volume of content within it, we opted to remove the modal and it's content from loading on small devices. Why won't the modal load in an Interchange file?

I'm on Foundation 5.4.6

**Comment by zurbrandon** [comment]
The forum may be able to help you out here: 

Maybe a wordpress path issue: 
http://foundation.zurb.com/forum/posts/493-interchange-with-wordpress

Maybe a reflow issue like here
http://foundation.zurb.com/forum/posts/2757-orbit--interchange

**Comment by gakimball** [comment]
Like @zurbrandon suggested, it's a reflow issue. When the Foundation JavaScript runs, it checks the entire page for plugins just once, right when the page loads. So any new HTML you add, like through Interchange, won't be recognized as plugins right away. Our reflow function will re-scan the page for any new plugins and then initialize them.

An Interchange element fires a `replace` event when a new piece of content is loaded. You can use this event to listen for when a modal is being dropped in.

```js
$('#interchangeElement').on('replace', function() {
});
```

Then use the below code to initialize new plugins.

```js
$(document).foundation('reveal', 'reflow');
```

This solution should work. Let me know if you have any questions! :)
